libretroshare/src/retroshare/rswire.h

```diff
@@ -73,26 +69,63 @@ class RsWirePlace
 
 };
 
+/************************************************************************
+ * Pulse comes in three flavours.
+ *
+ *
+ * Original Msg Pulse
+ * - Spontaneous msg, on your own group.
+ * - mPulseType = WIRE_PULSE_TYPE_ORIGINAL_MSG
+ * - Ref fields are empty.
+ *
+ * Reply to a Pulse (i.e Retweet), has two parts.
+ * as we want the retweet to reference the original, and the original to know about reply.
+ * This info will be duplicated in two msgs - but allow data to spread easier.
+ *
+ * Reply Msg Pulse, will be Top-Level Msg on Publisher's Group.
+ * - mPulseMode = WIRE_PULSE_TYPE_REPLY_MSG
+ * - Ref fields refer to Parent (InReplyTo) Msg.
+ *
+ * Reply Reference, is Child Msg of Parent Msg, on Parent Publisher's Group.
+ * - mPulseMode = WIRE_PULSE_TYPE_REPLY_REFERENCE
+ * - Ref fields refer to Reply Msg.
+ * - NB: This Msg requires Parent Msg for complete info, while other two are self-contained.
+ ***********************************************************************/
+
+#define  WIRE_PULSE_TYPE_ORIGINAL_MSG        (0x0001)
+#define  WIRE_PULSE_TYPE_REPLY_MSG           (0x0002)
+#define  WIRE_PULSE_TYPE_REPLY_REFERENCE     (0x0004)
+
+#define  WIRE_PULSE_TYPE_SENTIMENT_POSITIVE  (0x0010)
+#define  WIRE_PULSE_TYPE_SENTIMENT_NEUTRAL   (0x0020)
+#define  WIRE_PULSE_TYPE_SENTIMENT_NEGATIVE  (0x0040)
 
 class RsWirePulse
 {
 	public:
 
 	RsMsgMetaData mMeta;
 
-	std::string mPulseText; // all the text is stored here.
-	std::string mHashTags; 
-
-// These will be added at some point.
-//	std::string mInReplyPulse;
-
-//	uint32_t mPulseFlags;
-
-//	std::list<std::string> mMentions;
-//	std::list<std::string> mHashTags;
-//	std::list<std::string> mUrls;
-
-//	RsWirePlace mPlace;
+	// Store actual Pulse here.
+	std::string mPulseText;
+
+	uint32_t mPulseType;
+
+	// These Ref to the related (parent or reply) if reply (MODE_REPLY_MSG set)
+	// Mode                            REPLY_MSG only    REPLY_REFERENCE
+	RsGxsGroupId   mRefGroupId;   //   PARENT_GrpId      REPLY_GrpId
```

Comment:
Does that mean that when receives a  response to a tweet, the parent group of that tweet (meaning the twitter account?) needs to be subscribed to in order to see the msg. Consequently, all tweets in this new group that respond to other groups will in turn require to sync the other referred groups. I have the feeling that when you subscribe to a single group you'll recursively subscribe to everyone's.

Reply:
No - I've tried to design it to avoid this scenario.

A reply tweet, includes the information about the parent tweet. so you don't need to subscribe (or even have access to parent group/msgs) to understand the context of the tweet. You'll likely need to be subscribed to a group to reply though - as this is a gxs limitation.

Additionally, when replying a 2nd REF child msg is posted to the parent group/msg so that the original poster can track replies without having to subscribe to them either.

If you want to traverse further up or down the reply tree, then you'll need to subscribe to additional groups... But I've tried to design it so that users won't have to subscribe to thousands of groups.

I don't know how its going to work in practice, we'll have to run a trial and see!